boofuzz/sessions.py

```diff
@@ -51,30 +53,58 @@ class Target(object):
 
     :param connection: Connection to system under test.
     :type connection: itarget_connection.ITargetConnection
+    :param monitors: List of Monitors for this Target.
+    :type monitors: List[Union[IMonitor, pedrpc.Client]]
+    :param monitor_alive: List of Functions that are called when a Monitor is alive. It is passed
+                          the monitor instance that became alive. Use it to e.g. set options
+                          on restart.
```

Comment:
Yeah, good find. The system that I wanted to model should behave like c# events - basically, these are fancy callbacks that provide, amongst other things, a method to stop processing the event further (here: stopping the applying of callbacks).

However, I got one important thing wrong: the list of callbacks must be reversed, so that the last added callback is the first applied one for this to make any sense at all. For reasons you already mentioned, it would be wise to have the callback evaluate to a True-ish value to stop processing.

If this principle works for you, I'll update the function.

Reply:
I've now removed the function entirely, as it was only used in one place (as you mentioned in your comment below)